Lesson2.ipynb

```diff
@@ -648,6 +589,580 @@
+    "if \"Cherry\" in shoppingList:\n",
+    "    print(\"All good!\")\n",
+    "else:\n",
+    "    print(\"Don't forget the Cherries!\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "### Tuples\n",
+    "\n",
+    "Tuples are like Lists, with the small difference that they cannot be modified. We call them an **immutable** data structure.\n",
+    "\n",
+    "It is useful when you want to use a constant set of Data as reference.\n",
+    "\n",
+    "They are declared using parenthesis instead of square brackets for Lists, but all the operators or methods described earlier for Lists work for Tulpes too."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "weekDays = (\"Monday\", \"Tuesday\", \"Wednesday\", \"Thursday\", \"Friday\", \"Saturday\", \"Sunday\")\n",
+    "print(weekDays)\n",
+    "print(\"First day of the week:\", weekDays[0])\n",
+    "print(\"Working days:\", weekDays[0:5])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "Tuples are immutable. If we try to add or remove an element, an error is thrown."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "weekDays = (\"Monday\", \"Tuesday\", \"Wednesday\", \"Thursday\", \"Friday\", \"Saturday\", \"Sunday\")\n",
+    "weekDays.append(\"OtherDay\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "weekDays = (\"Monday\", \"Tuesday\", \"Wednesday\", \"Thursday\", \"Friday\", \"Saturday\", \"Sunday\")\n",
+    "weekDays.remove(\"Monday\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "But, we can cast a Tulpe to a List! We will then get a copy of the data that we can modify."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "weekDays = (\"Monday\", \"Tuesday\", \"Wednesday\", \"Thursday\", \"Friday\", \"Saturday\", \"Sunday\")\n",
+    "workingDays = list(weekDays)\n",
+    "workingDays.remove(\"Saturday\")\n",
+    "workingDays.remove(\"Sunday\")\n",
+    "\n",
+    "print(\"Working days:\", workingDays)\n",
+    "print(\"And the original Tulpe still exist and hasn't been modified:\", weekDays)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "### Dictionaries\n",
+    "\n",
+    "List and Tuples are nice but become quickly limited when we want to manipulate more detailed and organised data.\n",
+    "\n",
+    "Let's write down some users pesonal data for example:"
```

Comment:
"Let's write down some users pesonal data for example:" should read "Let's write down some users' personal data for example:"